### ISSUE TITLE
Zero shot classification 

### ISSUE THREAD

**Issue by narminGhaffari** [mlfoundations/open_clip#653]
Is there a possibility to use OpenClip model for zeroshot classification like the example which exsits in Clip original repository: 

import os
import clip
import torch
from torchvision.datasets import CIFAR100

# Load the model
device = "cuda" if torch.cuda.is_available() else "cpu"
model, preprocess = clip.load('ViT-B/32', device)

# Download the dataset
cifar100 = CIFAR100(root=os.path.expanduser("~/.cache"), download=True, train=False)

# Prepare the inputs
image, class_id = cifar100[3637]
image_input = preprocess(image).unsqueeze(0).to(device)
text_inputs = torch.cat([clip.tokenize(f"a photo of a {c}") for c in cifar100.classes]).to(device)

# Calculate features
with torch.no_grad():
    image_features = model.encode_image(image_input)
    text_features = model.encode_text(text_inputs)

# Pick the top 5 most similar labels for the image
image_features /= image_features.norm(dim=-1, keepdim=True)
text_features /= text_features.norm(dim=-1, keepdim=True)
similarity = (100.0 * image_features @ text_features.T).softmax(dim=-1)
values, indices = similarity[0].topk(5)

# Print the result
print("\nTop predictions:\n")
for value, index in zip(values, indices):
    print(f"{cifar100.classes[index]:>16s}: {100 * value.item():.2f}%")

**Comment by rom1504** [comment]
Yes, have a look at the first colab link in the readme

**Comment by narminGhaffari** [comment]
> t colab link in the readme
> […](#)

Thank you a lot @rom1504 . Just one more question. Is it possible also to use one of locally trained models (Checkpoints) instead of  `model, _, preprocess = open_clip.create_model_and_transforms('convnext_base_w', pretrained='laion2b_s13b_b82k_augreg') and if yes, how?
`
I really appreciate your help.
